ScoutSuite/providers/aws/facade/iam.py

```diff
@@ -0,0 +1,203 @@
+from ScoutSuite.providers.aws.facade.utils import AWSFacadeUtils
+from ScoutSuite.providers.aws.facade.basefacade import AWSBaseFacade
+from ScoutSuite.core.console import print_error, print_exception
+from ScoutSuite.providers.utils import run_concurrently, get_non_provider_id
+from ScoutSuite.providers.aws.utils import is_throttled
+
+
+class IAMFacade(AWSBaseFacade):
+    async def get_credential_reports(self):
+        client = AWSFacadeUtils.get_client('iam', self.session)
+        response = client.generate_credential_report()
+
+        if response['State'] != 'COMPLETE':
+            print_error('Failed to generate a credential report.')
+            return []
+
+        report = client.get_credential_report()['Content']
+
+        # The report is a CSV string. The first row contains the name of each column. The next rows
```

Comment:
Yea, I hesitated with this one, but since AWS's output is very sketchy (aka a CSV, for real?), I decided to at least parse the CSV. This way, the facade returns a nice object the caller can work with.